karapace/schema_reader.py

```diff
@@ -464,27 +476,93 @@ def _handle_msg_schema(self, key: dict, value: Optional[dict]) -> None:
         # won't interfere with the equality. Note: This means it is possible
         # for the REST API to return data that is formatted differently from
         # what is available in the topic.
+
+        parsed_schema: Optional[Union[Draft7Validator, AvroSchema, ProtobufSchema]] = None
+        resolved_dependencies: Optional[Dict[str, Dependency]] = None
+        if schema_type_parsed in [SchemaType.AVRO, SchemaType.JSONSCHEMA]:
+            try:
+                schema_str = json.dumps(json.loads(schema_str), sort_keys=True)
+            except json.JSONDecodeError:
+                LOG.error("Schema is not valid JSON")
+                return
+        elif schema_type_parsed == SchemaType.PROTOBUF:
+            try:
+                if schema_references:
+                    resolved_references = [
+                        Reference(reference["name"], reference["subject"], reference["version"])
+                        for reference in schema_references
+                    ]
+                    resolved_dependencies = self.resolve_references(resolved_references)
+                parsed_schema = parse_protobuf_schema_definition(
+                    schema_str,
+                    resolved_references,
+                    resolved_dependencies,
+                    validate_references=False,
+                )
+                schema_str = str(parsed_schema)
+            except InvalidSchema:
+                LOG.exception("Schema is not valid ProtoBuf definition")
```

Comment:
Is it intentional that the exceptions raised in `_resolve_reference` are propagated instead of just returning like `InvalidSchema` exceptions?

Reply:
No, we need to catch InvalidReferences there and log it with a return.